core/include/detray/tools/navigator.hpp

```diff
@@ -540,28 +528,7 @@ class navigator {
 
     private:
     /** the containers for all data */
-    detector_t detector;
+    detector_t *detector;
```

Comment:
Interesting. Do you need a non-const pointer to the detector? If so, maybe the class is not as const-correct yet as it should be...

Also, since you anyway have to touch all places where this variable is used: Could you give it a prefix that would suggest that it's a member variable? Most classes of this project use a `_` prefix, although I myself prefer `m_` because of the ATLAS conventions.

To keep with the current convention, just name it `_detector`. We can think about possibly harmonising the naming later.

Reply:
`const` seems possible. I changed it